server.R

```diff
@@ -154,13 +126,38 @@ shinyServer(function(input, output, session) {
   })
 
   observeEvent(input$update_confirm, {
-    if (input$update_confirm == TRUE) {
-      lapply(datatypes, function(x) {
-        updateSelectInput(session, paste0("dropdown_", x),
-          selected = input[[paste0("header_dropdown_", x)]]
-        )
-      })
-    }
+
+    req(input$update_confirm == TRUE)
+    isUpdateFolder(TRUE)
+    lapply(datatypes, function(x) {
+      updateSelectInput(session, paste0("dropdown_", x),
+        selected = input[[paste0("header_dropdown_", x)]]
+      )
+    })
+  })
+
+  ######## Update Folder List ########
+  lapply(c("header_dropdown_", "dropdown_"), function(x) {
+    observeEvent(ignoreInit = TRUE, input[[paste0(x, "project")]], {
+      # get synID of selected project
+      projectID <- datatype_list$projects[[input[[paste0(x, "project")]]]]
+
+      # gets folders per project
+      folder_list <- synapse_driver$getStorageDatasetsInProject(synStore_obj, projectID) %>% list2Vector()
+
+      # updates foldernames
+      updateSelectInput(session, paste0(x, "folder"), choices = sort(names(folder_list)))
+
+      if (x == "dropdown_") {
+        project_synID <<- projectID
+        datatype_list$folders <<- folder_list
+      }
+
+      req(isUpdateFolder() == TRUE)
+      # sync with header dropdown
+      updateSelectInput(session, "dropdown_folder", selected = input[["header_dropdown_folder"]])
+      isUpdateFolder(FALSE)
+    })
```

Comment:
sorry, just fixed some issues from the earlier codes.

> @rchai, I see - so lines 156-159 essentially take whatever the user selected in the header tab dropdown, and ensure the 'main' dropdowns (on the first tab of the app) are set to these values?

yes, these lines are used to force the "main" folder dropdown to update again with the same value in the "header" folder dropdown, if users click "confirm" button.